### ISSUE TITLE
REQ: Optimized Docker file

### ISSUE THREAD

**Issue by composer22** [signalfx/docker-collectd#11]
The base image is a pig.

I'd like to propose a cleanup

Here is my attempt at combined runs and purges, I'm not sure what else can be purged but my guess is there is additional crud:

Ubuntu 15.04 Base Image is 131.3 MB
SFX on top of that becomes 585 MB
Reorg RUNs became  532.1 MB
Add Autoclean: 510 MB
Purge 1 attempt: 359,6 MB
**Final Docker w/ purges 314.1 MB**

and here is the code:

```
# A Test refactor of Signal FX Dockerfile
# 
FROM ubuntu:15.04
MAINTAINER SignalFx Support <support+collectd@signalfx.com>

ENV DEBIAN_FRONTEND noninteractive
ENV EXPECTED_PLUGIN_VERSION 0.0.21
ENV EXPECTED_COLLECTD_VERSION 5.5.1.sfx0

RUN apt-get install -y apt-transport-https software-properties-common \
    && add-apt-repository ppa:signalfx/collectd-release \
    && add-apt-repository ppa:signalfx/collectd-plugin-release \
    && apt-get update \
    && apt-get -y upgrade \
    && apt-get install -y signalfx-collectd-plugin collectd jq \
    && grep "VERSION = \"$EXPECTED_PLUGIN_VERSION\"" /opt/signalfx-collectd-plugin/signalfx_metadata.py \
    && collectd -h | grep $EXPECTED_COLLECTD_VERSION \
    && rm -rf /etc/collectd \
    && apt-get --purge -y remove apt-transport-https openjdk-7-jre-headless perl perl-modules software-properties-common \
    && apt-get autoclean \
    && apt-get clean \
    && apt-get autoremove -y \
    && rm -rf /var/lib/apt/lists/* \
    && rm -rf /usr/share/man/* /usr/share/info/* \
    && rm -rf /usr/share/lintian/* /var/cache/man/*

# Setup our collectd
ADD configs /etc/
ADD run.sh /.docker/

WORKDIR /.docker/
CMD /.docker/run.sh
```

**Comment by composer22** [comment]
This is not to say it might impact others who are trying to work around this fatty image with their own attempt of purging in their own layers.

**Comment by mpetazzoni** [comment]
LGTM. We should do this. /cc @molner @udaysagar2177 @dgriff-tools @mdubbyap

**Comment by codesmith14** [comment]
I am working to include the docker plugin in the image by default.  I'll work on bringing these changes in as well.  Will update and close this issue, when complete.

**Comment by composer22** [comment]
Thanks 👍

**Comment by codesmith14** [comment]
Thank you @composer22 for your recommendations.  Over the past week I have included several of the changes you requested and added the docker collectd plugin.  We decided not to purge any dependencies which were installed with collectd or the SignalFx collectd plugin at this time.  Thank you for contributing these suggestions, they have been very helpful in keeping our base image size under control.
